community/kernel/src/main/java/org/neo4j/kernel/impl/store/LongerShortString.java

```diff
@@ -786,7 +787,9 @@ public static String decode(Bits bits)
             byte codePoint = bits.getByte( table.step );
             result[i] = table.decTranslate( codePoint );
         }
-        return String.valueOf(result);
+
+        // We know the char array is unshared, so use sharing constructor explicitly
+        return UnsafeUtil.newSharedArrayString( result );
```

Comment:
I was testing this code with the neo4j-root benchmark suite, and found that this made a positive difference. But there's no committed before/after comparison if that's what you're asking.